test/compilable/test21372.d

```diff
@@ -0,0 +1,13 @@
+// REQUIRED_ARGS: -de
+struct S
+{
+    deprecated void foo(T)(int) { }
+    void foo(T)(string) { }
```

Comment:
Please add another set of methods with reverse order.

```d
    void bar(T)(string) { }
    deprecated void bar(T)(int) { }
```
IIRC there were some problems caused by the ordering of overloads.

Reply:
Yes, the bug doesn't happen if you reverse the order, because of how chaining works. - That said, I can't recall an issue like this ever triggering off any but the first method.